util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java

```diff
@@ -140,10 +140,13 @@ public static ApacheHttpTransport createApacheHttpTransport(
         proxyUri != null || proxyCredentials == null,
         "if proxyUri is null than proxyCredentials should be null too");
 
-    HttpClientBuilder httpClientBuilder =
-        ApacheHttpTransport.newDefaultHttpClientBuilder()
-            .setProxy(
-                proxyUri == null ? null : new HttpHost(proxyUri.getHost(), proxyUri.getPort()));
+    HttpClientBuilder httpClientBuilder = ApacheHttpTransport.newDefaultHttpClientBuilder();
+
+    if (proxyUri != null) {
+      httpClientBuilder
+          .setRoutePlanner(null)
+          .setProxy(new HttpHost(proxyUri.getHost(), proxyUri.getPort()));
```

Comment:
Can we add a test here to ensure `setProxy` overrides `setRoutePlanner` for posterity

Reply:
I tried to add unit test but this is not trivial because proxy settings applied at very late stage of request execution. Probably we will need to do something like [this](https://stackoverflow.com/a/50590840/3227693), but I think that it will be better to track this as a separate AI.

I have tested this manually and it works, not sure about credentials though.